techbuzz/src/main/java/com/sivalabs/techbuzz/posts/web/controllers/AddVoteController.java

```diff
@@ -28,6 +38,48 @@ class AddVoteController {
     public ResponseEntity<Void> createVote(@Valid @RequestBody CreateVoteRequest request, @CurrentUser User loginUser) {
         var createVoteRequest = new CreateVoteRequest(request.postId(), loginUser.getId(), request.value());
         postService.addVote(createVoteRequest);
+
         return ResponseEntity.ok().build();
     }
+
+    @GetMapping(path = "/api/hx/votes/update/{voteType}/{postId}", produces = "text/html")
```

Comment:
Maybe we can use https://github.com/wimdeblauwe/htmx-spring-boot-thymeleaf library to simplify this.